crates/isograph_schema/src/validate_schema.rs

```diff
@@ -52,7 +52,7 @@ pub type ValidatedRefetchFieldStrategy = UseRefetchFieldRefetchStrategy<
 >;
 
 /// The validated defined field that shows up in the TScalarField generic.
-pub type ValidatedFieldDefinitionLocation = FieldType<ServerFieldId, ClientFieldId>;
+pub type ValidatedFieldDefinitionLocation = FieldType<ServerFieldId, ClientType<ClientFieldId>>;
```

Comment:
(2) let's make this part of the `ClientFieldSelectionLinkedFieldAssociatedData` not part of `ClientFieldSelectionScalarFieldAssociatedData`